icu4c/source/common/unicode/platform.h

```diff
@@ -493,13 +496,8 @@ namespace std {
  */
 #ifdef U_NOEXCEPT
     /* Use the predefined value. */
-#elif defined(_HAS_EXCEPTIONS) && !_HAS_EXCEPTIONS  /* Visual Studio */
-#   define U_NOEXCEPT
-#elif U_CPLUSPLUS_VERSION >= 11 || __has_feature(cxx_noexcept) || __has_extension(cxx_noexcept) \
-        || (defined(_MSC_VER) && _MSC_VER >= 1900)  /* Visual Studio 2015 */
-#   define U_NOEXCEPT noexcept
```

Comment:
I think VS2015 is still technically a supported platform? Though perhaps this could be reconsidered?

Reply:
I think VS 2015 is OK. It has MSC_VER==1900, so the original condition is met and U_NOEXCEPT is defined to be noexcept, which is what the new version does unconditionally.

Where things could differ is if _HAS_EXCEPTIONS == 0. I'm hoping it will be OK because we are restricting ourselves to compilers new enough to recognize noexcept as a keyword, which shouldn't depend on _HAS_EXCEPTIONS.

Reply:
Ah, good point. Yes, the define would be unchanged for VS2015. (Thanks).

I believe the macro `_HAS_EXCEPTIONS` is really only for the STL library though (possibly just the MSVC STL), so I don't think having it set or not would affect the core language itself.